File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const idxs = Array(priorities.length).fill().map((v,i)=>i);
    let maxVal = Math.max(...priorities);
    let answer = 0;
    
    while (priorities.length !== 0) {
        if (priorities[0] < maxVal) {
            priorities.push(priorities.shift());
            idxs.push(idxs.shift());
        } else {
            answer++;
            priorities.shift();
            if (idxs.shift() === location) return answer;
            maxVal = Math.max(...priorities);
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    q = deque()
    for i,n in enumerate(priorities):
        q.append((n,i))
        
    priorities.sort(reverse=True)
    
    curidx = 0
    while q:
        process = q.popleft()
        if process[0]==priorities[curidx]:
            if process[1]==location:
                return curidx+1
            curidx+=1
        else:
            q.append(process)
            